### DISCUSSION TITLE
No official support for the GitHub Actions -> GitHub Pages deployment workflow?

### DISCUSSION THREAD

**Discussion by seanh** [getpelican/pelican#3331] [General]
I noticed that the [GitHub Actions workflow for deploying a Pelican site to GitHub Pages](https://github.com/getpelican/pelican/blob/657ad64f49c219e1c3f7145088180a58d870a566/.github/workflows/github_pages.yml#L1-L66) has a comment on it saying "no official support."

I was wondering whether there's anything I can do to help the Pelican team to be able to provide official support for users of the workflow?

For example I could write up some notes explaining how the workflow works, step-by-step.

There's actually not much to it. All the workflow does is:

1. Checkout the user's Pelican site repo
2. Set up Python
3. Optionally checkout a theme repo as well (if <https://github.com/getpelican/pelican/pull/3330> is merged)
4. Configure GitHub Pages using [GitHub's official action for configuring GitHub Pages](https://github.com/actions/configure-pages)
5. Run `pip install`
6. Run `pelican` injecting the correct values for `SITEURL` and (if <https://github.com/getpelican/pelican/pull/3330> is merged) for `FEED_DOMAIN` as well
7. Deploy the compiled HTML to GitHub Pages using [GitHub's official action for deploying to GitHub Pages](https://github.com/actions/deploy-pages)

**Comment by seanh** [comment]
Alternatively, if the workflow isn't officially supported I wonder if it doesn't belong in the getpelican organization? I could copy the workflow to a separate repo in my own GitHub account, and send a PR to remove the workflow from the pelican repo and change the **Tips** page in the docs to just link to the workflow in my repo.

**Comment by justinmayer** [comment]
Thank you for the inquiry, Sean. More importantly, thank you for offering to help provide official support for users of the workflow, an offer that is graciously accepted.

As backstory, I don't believe any of the current maintainers use the publishing-to-GitHub-Pages-via-Actions deployment workflow, and therefore we are ill-equipped to provide support. Moreover, we are already struggling to keep up with the overall demands of open-source project management, and the prospect of adding yet another "support surface" is wildly infeasible. Ergo my "this is not officially supported" warning, which was only meant to set expectations for end users.

But your offer to help out does indeed change my perspective on this topic. I will remove that "unsupported" language that I added. I will also add you as a [code owner](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) for this workflow so that PRs that touch that file will generate review requests for you.

Thanks again for the thoughtful discussion and the kind offer to help out with the project. Much appreciated! 😁

**Reply by seanh** [reply]
Great, thanks!

FYI I get a lot of GitHub notifications (since I use it for my day job) so feel free to ping me if I miss a PR.

Hopefully there won't be too much more work to do to maintain this workflow. I missed a few things on the first pass (the ability to clone a theme repo; setting `FEED_DOMAIN`, etc). But I'm hoping (perhaps naively) that it's more or less complete now.

**Reply by justinmayer** [reply]
I removed the "unsupported" language via: 05535c7d
